service/storage_service.cc

```diff
@@ -559,7 +560,13 @@ future<> storage_service::topology_state_load() {
     for (const auto& e: tmptr->get_all_endpoints()) {
         const auto ep = tmptr->get_endpoint_for_host_id(e);
         if (!is_me(e) && !_gossiper.get_endpoint_state_ptr(ep)) {
-            co_await _gossiper.add_saved_endpoint(ep);
+            gms::loaded_endpoint_state st;
+            const auto& topo = tmptr->get_topology();
+            const auto* node = topo.find_node(e);
```

Comment:
Can node be nullptr here?

Reply:
It'd be a bug.
If it's in token_metadata then it must be in topology.  They are tightly coupled.  That's why topology is a member of t.m. and is manipulated through it, rather than being a standalone class, like raft topology.

Reply:
I added a comment explaining this in next version